docs/modules/ROOT/pages/adr/0043-appcat-framework-production-architecture.adoc

```diff
@@ -0,0 +1,1750 @@
+Service restoration supports both automatic restore (user-initiated via XRD) and manual restore (operator-triggered for complex recovery scenarios).
+
+**Automatic Restore (User-Initiated):**
+
+Some services support automatic restoration through the XRD specification, allowing Service Users to create a new instance that automatically restores from a broken instance's backup.
+The user specifies the source instance and backup snapshot in the restore parameters, and the composition function orchestrates the restoration during instance creation.
+
+[source]
+----
+1. Service User creates new instance with restore specification
+   → Specifies source instance name in restore parameters
+   → Optionally selects specific backup snapshot
+   → Composition function detects restore request
+
+2. Composition function orchestrates restoration
+   → Locates backup artifacts from source instance
+   → Provisions new instance infrastructure
+   → Restores data from backup during initialization
+
+3. Instance becomes ready
+   → Service User accesses restored instance
+   → Original broken instance remains for investigation
+----
+
+**Manual Restore (Operator-Triggered):**
+
+For complex recovery scenarios requiring direct intervention, Service Operators manually trigger restoration by creating restore resources that target specific backup snapshots and restoration methods.
+
+[source]
+----
+1. Incident detected
+   → Service Operator verifies issue through metrics and logs
+   → Confirms data corruption or loss requiring restoration
+
+2. Identifies recovery point
+   → Lists available backup snapshots
+   → Selects snapshot from before incident
+
+3. Initiates restoration
+   → Scales down workload to prevent writes
+   → Creates restore resource targeting selected snapshot
+   → Backup system restores data from object storage
+
+4. Validates restoration
+   → Scales up workload
+   → Verifies data integrity through metrics and smoke tests
+   → Notifies Service User of resolution
+----
+
+=== Framework Engineer
+
+==== Persona Description
+
+*Framework Engineer* maintains platform infrastructure, composition functions, KCL framework, CI/CD, and testing infrastructure.
+
+==== Requirements
+
+1. *Maintain Composition Function*: Generic Go function handling all services
+2. *Maintain KCL Framework*: Type schemas, distributions, profiles
+3. *Manage Platform*: Crossplane, providers, ArgoCD, AppCat components and other 3rd Operators such as CNPG
+4. *CI/CD Pipeline*: Automated testing, building, deployment
+5. *Documentation*: ADRs, runbooks, architecture diagrams
+
+==== Solution
+
+===== 1. Composition Function Architecture
+
+**Core Logic Flow:**
+
+Framework Engineer maintains the generic composition function that:
+
+1. Receives ConfigMap reference from Composition `inputRef` and loads service configuration
+2. Extracts composite parameters from Crossplane request and binds to generated Go types
```

Comment:
Also as already mentioned: let's keep types out of the go code.

Reply:
Do you also mean the XRD as well?

Reply:
Yes. We can access any fields of any k8s objects easily with Crossplane's fieldpath library: https://pkg.go.dev/github.com/crossplane/crossplane-runtime/pkg/fieldpath

This would also play nice with the templating, as the service maintainer could specify simple fieldpaths to access composite fields (or any field from any resource in the state) in the KCL.

Reply:
types are out, will use Crossplane field paths to navigate the function.